subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1298,6 +1302,7 @@ int ztls_accept_ctx(struct net_context *parent, struct sockaddr *addr,
 	}
 
 	net_context_set_accepting(child, false);
+	net_context_ref(child);
```

Comment:
Could you check the error paths also below this line, doing ref here might cause some leaks in case of errors.

Reply:
The error handler (underneath `error:` label) calls socket close function which calls `net_context_put`. In this function, we have an explicit comment:
```
	/* Decrement refcount on user app's behalf */
	net_context_unref(context);
```
Unless I miss something, we're on the safe side if `net_context_put` is called and not furthe action is needed. At least no other action is taken in "typical" socket close operation.